hoomd/md/pytest/test_pair.py

```diff
@@ -0,0 +1,751 @@
+        snap.particles.diameter[1] = 0.5
+
+
+def test_attached_params(simulation_factory, lattice_snapshot_factory,
+                         valid_params):
+    pair_potential, pair_potential_dict, extra_args = valid_params
+    pair_keys = valid_params.pair_potential_params.keys()
+    particle_types = list(set(itertools.chain.from_iterable(pair_keys)))
+    pot = valid_params.pair_potential(**valid_params.extra_args,
+                                      nlist=hoomd.md.nlist.Cell(),
+                                      r_cut=2.5, mode='none')
+    for pair in valid_params.pair_potential_params:
+        pot.params[pair] = valid_params.pair_potential_params[pair]
+
+    snap = lattice_snapshot_factory(particle_types=particle_types,
+                                    n=10, a=1.5, r=0.01)
+
+    _update_snap(valid_params.pair_potential, snap)
+    if snap.exists:
+        snap.particles.typeid[:] = np.random.randint(0,
+                                                     len(snap.particles.types),
+                                                     snap.particles.N)
+    sim = simulation_factory(snap)
+    sim.operations.integrator = hoomd.md.Integrator(dt=0.005)
+    sim.operations.integrator.forces.append(pot)
+    sim.run(10)
+    _assert_equivalent_type_params(pot.params.to_dict(),
+                                   valid_params.pair_potential_params)
+
+
+def test_run(simulation_factory, lattice_snapshot_factory, valid_params):
+    pair_keys = valid_params.pair_potential_params.keys()
+    particle_types = list(set(itertools.chain.from_iterable(pair_keys)))
+    pot = valid_params.pair_potential(**valid_params.extra_args,
+                                      nlist=hoomd.md.nlist.Cell(),
+                                      r_cut=2.5, mode='none')
+    for pair in valid_params.pair_potential_params:
+        pot.params[pair] = valid_params.pair_potential_params[pair]
+
+    snap = lattice_snapshot_factory(particle_types=particle_types,
+                                    n=7, a=1.7, r=0.01)
+    if snap.exists:
+        snap.particles.typeid[:] = np.random.randint(0,
+                                                     len(snap.particles.types),
+                                                     snap.particles.N)
+    sim = simulation_factory(snap)
+    integrator = hoomd.md.Integrator(dt=0.005)
+    integrator.forces.append(pot)
+    integrator.methods.append(hoomd.md.methods.Langevin(hoomd.filter.All(),
+                                                        kT=1, seed=1))
+    sim.operations.integrator = integrator
+    sim.operations.schedule()
+    for nsteps in [3, 5, 10]:
+        old_snap = sim.state.snapshot
+        sim.run(nsteps)
+        new_snap = sim.state.snapshot
+        if new_snap.exists:
+            with pytest.raises(AssertionError):
+                np.testing.assert_allclose(new_snap.particles.position,
+                                           old_snap.particles.position)
```

Comment:
Just assert that they are not all close. We should use `pytest.raises` for an `AssertionError` just use NumPy's `allclose` function and assert not that.